Add allowed_cilogon list functionality (idphint short list)

juleengraham opened the pull request.

juleengraham committed:
Add allowed_cilogon list functionality

dannon commented on lib/galaxy/webapps/galaxy/controllers/authnz.py:
```diff
@@ -136,3 +136,10 @@ def get_cilogon_idps(self, trans, **kwargs):
+
+    @web.expose
+    @web.json
+    def get_allowed_cilogon_idps(self, trans, **kwargs):
```
Instead of the additional endpoint (and request to fetch it), let's just apply the filtering in `get_cilogon_idps` above.

dannon commented on lib/galaxy/authnz/managers.py:
```diff
@@ -157,12 +157,19 @@ def _parse_custos_config(self, config_xml):
             rtv['idphint'] = config_xml.find('idphint').text
+        # can prob get rid of idphint ^
+        if config_xml.find('allowed_idps') is not None:
+            self.allowed_idps = config_xml.find('allowed_idps').text
```
Can you add an example to the sample for this?

juleengraham replied:
@dannon do you mean adding an example in the docs or adding a comment here of what it would look like?

juleengraham committed:
Remove extra api request, validate entity ids, filter out blocked idps

juleengraham committed:
add validation of idphint at login

juleengraham committed:
validate idps, switch the allow

switch allowed_idps in configs to separate allow_idp tags for each institution, update oidc_backend sample

juleengraham committed:
pre-select idphint if only one allowed idp

juleengraham committed:
Note about allow list for idps if both Custos and CIlogon are enabled

juleengraham committed:
consistency: change 8000 to 8080
